doc-content/enterprise-only/optimizer/ex-nQueens-ref.adoc

```diff
@@ -0,0 +1,164 @@
+[id='ex-nQueens-ref']
+= N Queens
+
+Place _n_ queens on a _n_ sized chessboard so that no two queens can attack each other.
+The most common _n_ queens puzzle is the eight queens puzzle, with __n = 8__:
+
+image::UseCasesAndExamples/NQueens/nQueensScreenshot.png[align="center"]
+
+Constraints:
+
+* Use a chessboard of _n_ columns and _n_ rows.
+* Place _n_ queens on the chessboard.
+* No two queens can attack each other. A queen can attack any other queen on the same horizontal, vertical or diagonal line.
+
+This documentation heavily uses the four queens puzzle as the primary example.
+
+A proposed solution could be:
+
+.A Wrong Solution for the Four Queens Puzzle
+image::UseCasesAndExamples/NQueens/partiallySolvedNQueens04Explained.png[align="center"]
+
+The above solution is wrong because queens `A1` and `B0` can attack each other (so can queens `B0` and ``D0``). Removing queen `B0` would respect the "no two queens can attack each other" constraint, but would break the "place _n_ queens" constraint.
+
+Below is a correct solution:
+
+.A Correct Solution for the Four Queens Puzzle
+image::UseCasesAndExamples/NQueens/solvedNQueens04.png[align="center"]
+
+All the constraints have been met, so the solution is correct. 
+
+Note that most _n_ queens puzzles have multiple correct solutions.
+We will focus on finding a single correct solution for a given _n_, not on finding the number of possible correct solutions for a given _n_.
+
+
+[[nQueensProblemSize]]
+.Problem Size
+[source,options="nowrap"]
+----
+4queens   has   4 queens with a search space of    256.
+8queens   has   8 queens with a search space of   10^7.
+16queens  has  16 queens with a search space of  10^19.
+32queens  has  32 queens with a search space of  10^48.
+64queens  has  64 queens with a search space of 10^115.
+256queens has 256 queens with a search space of 10^616.
+----
+
+The implementation of the _n_ queens example has not been optimized because it functions as a beginner example. Nevertheless, it can easily handle 64 queens.
+With a few changes it has been shown to easily handle 5000 queens and more.
+
+
+[[nQueensDomainModel]]
+.Domain Model
```

Comment:
Sub-heading, not label. Could be it's own module (hence why it's anchored), but for the sake of time, sub-heading within the module is fine for now:
```
== N Queens domain model

The N Queens example uses...
```

Reply:
I am definitely lost in why you think some things should be sub-headings instead of labels. I made everything labels, well that was the intention, one or two slipped through.

Reply:
Per mode doc convention, headings are for standalone, typically non-repeated content, such as concepts, tasks, and references. Usually they're their own modules, but concepts and refs can technically contain sub-headings too, but I've found that almost always a sub-headed section can be its own module as well that's just included. Labels are used for elements like "Procedure," "Prerequisites," "Additional resources," "Table," "Figure" (the last two automatically placed by asciidoc), or code snippet labels. Labeled content is typically repeated types of content within headed sections, but not always (like code snippet labels).

In short, you use labels to label a block of content, not to head a section. This example is clearly a section of content, not a block to be labeled, and thus makes more sense as a sub-heading.

Reply:
I have attempted to implement what you write here, which resulted in "Domain model" being a heading in some cases and a section in others. When I push the changes (won't reupload right now) please review; we might have to leave it be right now?

Reply:
You can leave anything be right now that you wish. We are admittedly at the end of the road, and that's okay. That's the purpose of Agile, to get things out and iterate to improve. Can't do everything perfectly every time always.